### ISSUE TITLE
JCodeModel.parseType(String) silently ignores type params in specific scenarios

### ISSUE THREAD

**Issue by perceptron8** [eclipse-ee4j/jaxb-ri#1505]
If there are multiple type params and (at least) one one of them is generic, `parseType` works properly only if this param is the last one.

Because of that this test passes:
```java
JCodeModel model = new JCodeModel();
JType type = model.parseType("Map<K,Pair<X,Y>>");
assertEquals("Map<K,Pair<X,Y>>", type.fullName());
```

This one fails miserably:
```java
JCodeModel model = new JCodeModel();
JType type = model.parseType("Map<Pair<X,Y>,V>");
assertEquals("Map<Pair<X,Y>,V>", type.fullName());
```

```
org.opentest4j.AssertionFailedError: expected: <Map<Pair<X,Y>,V>> but was: <Map<Pair<X,Y>>>
```

Shouldn't `TypeNameParser` be smart enough to deal with such simple names?

**Comment by perceptron8** [comment]
In the meantime, I'd like to assure you, that I know how to construct these types programmatically and confirm, that my use case is indeed strictly related to parsing.

```java
JCodeModel model = new JCodeModel();
JClass Map = model.ref("Map");
JClass Pair = model.ref("Pair");
JClass X = model.ref("X");
JClass Y = model.ref("Y");
JClass V = model.ref("V");
JType type = Map.narrow(Pair.narrow(X, Y), V);
assertEquals("Map<Pair<X,Y>,V>", type.fullName());
```

```java
JCodeModel model = new JCodeModel();
JClass Map = model.ref("Map");
JClass Pair = model.ref("Pair");
JClass K = model.ref("K");
JClass X = model.ref("X");
JClass Y = model.ref("Y");
JType type = Map.narrow(K, Pair.narrow(X, Y));
assertEquals("Map<K,Pair<X,Y>>", type.fullName());
```

**Comment by lukasj** [comment]
should be fixed in master (4.0.0)

**Comment by perceptron8** [comment]
Thank you very much!
I apologize for not writing sooner.
